torch/_dynamo/backends/common.py

```diff
@@ -59,7 +59,13 @@ def __call__(self, gm: torch.fx.GraphModule, example_inputs, **kwargs):
 
         # NB: dont delete counter increment
         counters["aot_autograd"]["total"] += 1
-        use_fallback = False
+
+        import torch.distributed.distributed_c10d as c10d
+        # Fall back to not enable autograd if mutation has to be supported.
```

Comment:
Do the tests you added fail due to this? It feels like this is just a DCE problem? It should work to run AOTAutograd here.

Reply:
The issue was mutations of the input parameter in the recv function. Because this mutation violates the functional paradigm assumption for functions, the parameter fails to be modified if the Autograd is enabled, due to the use of functional objects. I am not aware if there is a way around it, except if we adopt an alternative functional API for the recv function.